### ISSUE TITLE
Call to a member function ajax() on string

### ISSUE THREAD

**Issue by Hossien-Salamhe** [yajra/laravel-datatables#2351]
How i can solve this problem?

Call to a member function ajax() on string

$this->request  in \Yajra\DataTables\Services\DataTable::request return "datatables.request"

```php
public function request()
    {
        return $this->request ?: $this->request = resolve('datatables.request');
    }
```
```php
if ($this->request()->ajax() && $this->request()->wantsJson()) {
            return app()->call([$this, 'ajax']);
        }
```

```php
public function index(OrdersDataTable $ordersDataTable, OrderIndex $data)
    {
        return $ordersDataTable->render('barber.dynamic.index', ['data' => $data]);
    }

```

- ubuntu 18.04.4
- PHP Version 7.4
- Laravel Version 6.2
- Laravel-Datatables Version 9.0

**Comment by yajra** [comment]
Can you post your full OrdersDataTable class or some snippets to reproduce the issue? Ajax method is already defined in the base class https://github.com/yajra/laravel-datatables-buttons/blob/4.0/src/Services/DataTable.php#L187.

**Comment by Hossien-Salamhe** [comment]
Here you are!

```php
<?php

namespace App\DataTables\Barber;

use App\Models\Order;
use Yajra\DataTables\DataTableAbstract;
use Yajra\DataTables\Html\Builder;
use Yajra\DataTables\Services\DataTable;

class OrdersDataTable extends DataTable
{
    public function dataTable($query)
    {
        return datatables($query)
            ->addColumn('actions', function ($order) {
                return view('common.partials.btn.actions', [
                    'editUrl' => action('Barber\OrderController@edit', $order)
                ])->render();
            })->rawColumns([
                'actions'
            ]);
    }

    public function query()
    {
        return Order::with('service', 'user')->orderBy('created_at', 'desc');
    }

    public function html()
    {
        return $this->builder()
            ->columns($this->getColumns())
            ->minifiedAjax()
            ->parameters($this->getBuilderParameters());
    }

    protected function getColumns()
    {
        return [
            [
                'name' => 'id',
                'data' => 'id',
                'title' => '#',
            ], [
                'name' => 'service.title',
                'data' => 'service.title',
                'title' => trans('admin.service'),
            ], [
                'name' => 'start_time',
                'data' => 'start_time',
                'title' => trans('admin.start_time'),
            ], [
                'name' => 'user.name',
                'data' => 'user.name',
                'title' => trans('admin.user'),
            ], [
                'name' => 'status',
                'data' => 'status',
                'title' => trans('admin.status'),
//                'visible' => false,
            ], [
                'name' => 'payment_type',
                'data' => 'payment_type',
                'title' => trans('admin.payment_type'),
                'visible' => false,
            ], [
                'name' => 'actions',
                'data' => 'actions',
                'title' => trans('admin.actions'),
                'exportable' => false,
                'printable' => false,
                'orderable' => false,
                'searchable' => false,
            ],

        ];
    }

    protected function filename()
    {
        return 'Orders_' . date('YmdHis');
    }
}

```

 the problem here https://github.com/yajra/laravel-datatables-buttons/blob/4.0/src/Services/DataTable.php#L157

``` $this->request``` is returned this string ``` "datatables.request"```

problem exactly here ```$this->request()->ajax()``` Call to a member function ajax() on string

the request function returned a string value
```php
public function request()
    {
        return $this->request ?: $this->request = resolve('datatables.request');
    }
```

**Comment by henryfuentes** [comment]
+1

**Comment by yajra** [comment]
@rodrigosoareschaves thank you for pointing it out. Maybe we should use app instead of resolve? Can you verify if that would fix the issue?

```php
resolve('datatables.request');
// to
app('datatables.request');
```

**Comment by rodrigosoareschaves** [comment]
Hi @yajra ! Pleasure to meet you, and glad to help!
Great idea!
Seems that is working good with the "app" function instead of "resolve" function. I changed this part of the code:
```
    public function request()
    {
        return $this->request ?: $this->request = app('datatables.request');
    }
```
directly on  "vendor/yajra/laravel-datatables-buttons/src/Services/DataTable.php" line 179. 
And removed the ugly code on my "public/index.php" added on past comment. It's working better. Thank you

**Comment by yajra** [comment]
@rodrigosoareschaves thanks for confirming, fixed on buttons package v4.9.2.
